File: xamarin/Samples/Showcase-iOS/ViewController/AudioDisplayViewController.cs
using Foundation;
using LiferayScreens;
using System;
using UIKit;

namespace ShowcaseiOS.ViewController
{
    public partial class AudioDisplayViewController : UIViewController, IFileDisplayScreenletDelegate
    {
        public AudioDisplayViewController(IntPtr handle) : base(handle) { }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            this.audioDisplayScreenlet.ClassPK = 57433;
            this.audioDisplayScreenlet.ClassName = "com.liferay.document.library.kernel.model.DLFileEntry";

            this.audioDisplayScreenlet.Delegate = this;
        }

        /* IFileDisplayScreenletDelegate */

        [Export("screenlet:onFileAssetError:")]
        public virtual void OnFileAssetError(FileDisplayScreenlet screenlet, NSError error)
        {
            Console.WriteLine($"Audio display failed: {error.DebugDescription}");
        }

        [Export("screenlet:onFileAssetResponse:")]
        public virtual void OnFileAssetResponse(FileDisplayScreenlet screenlet, NSUrl url)
        {
            Console.WriteLine($"Audio display success: {url.DebugDescription}");
        }
    }
}

File: ios/Samples/Showcase-ObjectiveC/Showcase-ObjectiveC/ViewControllers/Files/AudioDisplayViewController.m
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */


@import LiferayScreens;
#import "LiferayLogger.h"
#import "AudioDisplayViewController.h"

@interface AudioDisplayViewController () <FileDisplayScreenletDelegate>

@property (weak, nonatomic) IBOutlet AudioDisplayScreenlet *screenlet;

@end

@implementation AudioDisplayViewController

- (void)viewDidLoad {
	[super viewDidLoad];

	self.screenlet.delegate = self;
	self.screenlet.presentingViewController = self;
	self.screenlet.classPK = [LiferayServerContext longPropertyForKey:@"audioDisplayClassPK"];
}

#pragma mark FileDisplayScreenletDelegate

- (void)screenlet:(FileDisplayScreenlet *)screenlet onFileAssetError:(NSError *)error {
	LiferayLog(error.debugDescription);
}

- (void)screenlet:(FileDisplayScreenlet *)screenlet onFileAssetResponse:(NSURL *)url {
	LiferayLog(url);
}

@end
